File: behaviorial/memento/python/originator.py
from memento import GameState


class Game:
    def __init__(self, level, health):
        self.level = level
        self.health = health

    def save(self):
        return GameState(self.level, self.health)

    def restore(self, state):
        self.level = state.level
        self.health = state.health

    def display(self):
        return f"Level: {self.level}, Health: {self.health}"

File: behaviorial/memento/typescript/originator.ts
import { GameState } from './memento';

export class Game {
	level: number;
	health: number;

	constructor(level: number, health: number) {
		this.level = level;
		this.health = health;
	}

	save(): GameState {
		return new GameState(this.level, this.health);
	}

	restore(state: GameState): void {
		this.level = state.level;
		this.health = state.health;
	}

	display(): string {
		return `Level: ${this.level}, Health: ${this.health}`;
	}
}
